File: Linguagens de Programação/GOLang/oddUntilN.go
package main
import ("fmt")

func main() {
    fmt.Println("Impressora de números impares até N")
	fmt.Println("Digite o n: ")
    var n int 
    fmt.Scanln(&n)
	var i int = 0
	for i < n {
		if i % 2 != 0 {
			fmt.Println(i)
			i++
		}
	}
}

File: Linguagens de Programação/Haskell/oddUntilN.hs
oddNumbers :: Int -> [Int]
oddNumbers n = take n [x | x <- [1,3..]]

main = do
    putStrLn "Digite um número inteiro positivo:"
    n <- readLn
    let result = oddNumbers n
    putStrLn ("Os primeiros " ++ show n ++ " números ímpares são: " ++ show result)
